pygfx/renderers/wgpu/wgsl/points.wgsl

```diff
@@ -313,6 +313,75 @@ fn fs_main(varyings: Varyings) -> FragmentOutput {
 }
 
 
+fn get_signed_distance_to_pygfx_outer_triangle(coord_in: vec2<f32>, size: f32) -> vec2<f32> {
+    let m_pi = 3.141592653589793;
+    let m_sqrt_2 = 1.4142135623730951;
+    let half_size = size * 0.5;
+
+    // The coords below assume OpenGL coords, but y is flipped, so we flip the coords.
+    var coord = vec2<f32>(coord_in.x, -coord_in.y);
+
+    // https://math.stackexchange.com/a/4073070
+    // equilateral triangle has length of size
+    //    sqrt(3) - 1
+    let triangle_x = (sqrt(3.) - 1.);
+    let one_minus_triangle_x = 2. - sqrt(3.);
+    let triangle_length = m_sqrt_2 * triangle_x;
+
+    let pygfx_width = 0.10;
+
+    // Offset the coordinates so we can take up the maximum of the quad
+    let coord_1 = coord / size + vec2<f32>(0.5);
+
+    let v1 = normalize(vec2<f32>(one_minus_triangle_x, -1));
+    let r1_out = dot(coord_1, v1);
+    let r1_in  = r1_out + pygfx_width;
+    let r1 = max(r1_out, -r1_in);
+
+    let v2 = normalize(vec2<f32>(-1, one_minus_triangle_x));
+    let r2_out = dot(coord_1, v2);
+    let r2_in  = r2_out + pygfx_width;
+    let r2 = max(r2_out, -r2_in);
+
+    let v3 = normalize(vec2<f32>(triangle_x, triangle_x));
+    let r3_out = dot(coord_1 - vec2(1, one_minus_triangle_x), v3);
+    let r3_in  = r3_out + pygfx_width;
+    let r3 = max(r3_out, -r3_in);
+
+    // return max(max(r1, r2), r3);
+    let outer_edge_triangle = max(
+        max(r1_out, max(r2_out, r3_out)),
+        -max(r1_in, max(r2_in, r3_in))
+    );
+
+
+    let inner_offset = 0.5 * (triangle_length - pygfx_width / 2.);
+    let r1_out_blue = -r1_out - inner_offset;
+    let r1_in_blue = r1_out_blue + pygfx_width;
+    let r1_blue = max(
+        max(r2_out, r3_in),
+        max(r1_out_blue, -r1_in_blue)
+    );
+
+    let r2_out_blue = -r2_out - inner_offset;
+    let r2_in_blue = r2_out_blue + pygfx_width;
+    let r2_blue = max(
+        max(r3_out, r1_in),
+        max(r2_out_blue, -r2_in_blue)
+    );
+
+    let r3_out_blue = -r3_out - inner_offset;
+    let r3_in_blue = r3_out_blue + pygfx_width;
+    let r3_blue = max(
+        max(r1_out, r2_in),
+        max(r3_out_blue, -r3_in_blue)
+    );
+
+    let inner_triangle = min(r1_blue, min(r2_blue, r3_blue));
+
+    return vec2<f32>(outer_edge_triangle, inner_triangle) * size;
```

Comment:
I found it strange that your original intersections and unions were written in "scale aware" coordinates". 

It was easier for me to write it in these "normalized units" and only think of the scale at the very end.

Reply:
The code for most markers was copied from Rougier's paper and adjusted for wgsl. TBH I have not attempted creating my own marker yet. I guess that makes you officially the expert on this code ;)